PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/SingleOperationInLoopCheck.java

```diff
@@ -0,0 +1,288 @@
+import com.intellij.psi.PsiMethodCallExpression;
+import com.intellij.psi.PsiStatement;
+import com.intellij.psi.PsiVariable;
+import com.intellij.psi.PsiWhileStatement;
+import com.intellij.psi.util.PsiTreeUtil;
+
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+
+/**
+ * Inspection to check if there is a Text Analytics client operation inside a loop.
+ * If a Text Analytics client operation is found inside a loop,
+ * and the API has a batch alternative, a problem will be registered.
+ *
+ * THis is an example of a situation where the inspection should register a problem:
+ *
+ * // Loop through the list of texts and detect the language for each text
+*         1. for (String text : texts) {
+*             DetectedLanguage detectedLanguage = textAnalyticsClient.detectLanguage(text);
+*             System.out.println("Text: " + text + " | Detected Language: " + detectedLanguage.getName() + " | Confidence Score: " + detectedLanguage.getConfidenceScore());
+*         }
+*
+*         // Loop through the list of texts and detect the language for each text
+*        2.  DetectedLanguage detectedLanguage = null;
+*         for (String text : texts) {
+*             textAnalyticsClient.detectLanguage(text);
+ */
+public class SingleOperationInLoopCheck extends LocalInspectionTool {
+
+    /**
+     * Build the visitor for the inspection. This visitor will be used to traverse the PSI tree.
+     * @param holder The holder for the problems found
+     * @param isOnTheFly Whether the inspection is running on the fly. If true, the inspection is running as you type.
+     * @return The visitor for the inspection
+     */
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+        return new SingleOperationInLoopVisitor(holder, isOnTheFly);
+    }
+
+    /**
+     * Visitor class to traverse the PSI tree and check for single Azure client operation inside a loop.
+     * The visitor will check for loops of type for, foreach, while, and do-while.
+     * The visitor will check for a Text Analytics client operation inside the loop.
+     * If a Text Analytics client operation is found inside the loop, and the API has a batch alternative, a problem will be registered.
+     */
+    public static class SingleOperationInLoopVisitor extends JavaElementVisitor {
+
+        // Define the holder for the problems found and whether the inspection is running on the fly
+        private static ProblemsHolder holder = null;
+        private final boolean isOnTheFly;
+
+        // Define the logger for the visitor
+        private static final Logger LOGGER = Logger.getLogger(SingleOperationInLoopCheck.class.getName());
+
+        // Define the suggestion message for the problem
+        private static String SUGGESTION = "";
```

Comment:
String in Java is immutable. We don't need to assign a value first and then change it.